### ISSUE TITLE
Add Messaging Support to Toolbox XML

### ISSUE THREAD

**Issue by vicng** [google/blockly#1014]
Toolbox XML currently doesn't support the use of message localization like in JSON block definitions.

eg.
```
<category name="%{BKY_TOOLBOX_MATH_TITLE}" colour="%{BKY_COLOUR_HUE}">
</category>
```

This should be relatively easy to support. The only question is how to do it. Here are a couple of options:
1) Add another attribute to category for display purposes (ie. `"label"`). If present, use this value as the display name. Otherwise, just show whatever is in `"name"`.
2) Just allow the `"name"` attribute to localize values. However, this might break any existing code that references a category by a specific value.

**Comment by RoboErikG** [comment]
I'd lean towards 2. I don't think 2 breaks anything on our side as we use the custom attribute to handle special categories. If any devs are looking for a specific name they probably aren't doing i18n, so they could just keep using a string literal.

For reference, Blockly Games currently [uses soy templates](https://github.com/google/blockly-games/search?utf8=%E2%9C%93&q=category&type=) to replace all the category names.
<category name="{msg meaning="Games.turtle" desc="IBID"}Turtle{/msg}">

**Comment by AnmAtAnm** [comment]
This was implemented long ago, and #1929 also added support for localizing the buttons and labels.
